spec/lib/voight_kampff/test_spec.rb

```diff
@@ -27,14 +29,18 @@
   end
 
   context 'after the first run' do
-    before { VoightKampff::Test.new('anything').bot? }
-
-    it 'is fast' do
-      expect(
-        Benchmark.realtime do
-          20.times { VoightKampff::Test.new('anything').bot? }
-        end
-      ).to be < 0.003
+    def time_of_run
+      Benchmark.realtime { VoightKampff::Test.new('anything').bot? }
+    end
+
+    let(:number_of_runs) { 20 }
+
+    times_faster = 2
+
+    it "is at least #{times_faster} times faster" do
+      expect(time_of_run / times_faster).to be > (
+        (1..number_of_runs).map { time_of_run }.sum / number_of_runs
+      )
```

Comment:
I noticed this was failing on CI in some cases. So it seems it will have to be changed a bit. I also like the idea of having some hard-coded time in here so we're sure it doesn't get too slow. Maybe it would be best to change the new test to `1.5` times faster or so. And then keep the old test as well, just as a check to make sure performance doesn't become worse over time.

Reply:
I found the problem: the test itself (`bundle exec rspec spec/lib/voight_kampff/test_spec.rb:31`) is passing, but all tests (`bundle exec rspec`) are failing here. We're testing performance in the `VoightKampff::Test.crawler_regexp`, which is caching into `@crawler_regexp` variable, so, I guess that some another test(s) fill it before this test, and the first run is already cached and optimized. I've added nilifying of `@crawler_regexp` before this test and it's passing in all set.